spring-session-jdbc/src/main/java/org/springframework/session/jdbc/JdbcOperationsSessionRepository.java

```diff
@@ -757,7 +800,7 @@ public void setAttribute(String attributeName, Object attributeValue) {
 								? oldDeltaValue
 								: DeltaValue.UPDATED);
 			}
-			this.delegate.setAttribute(attributeName, attributeValue);
+			this.delegate.setAttribute(attributeName, constantSupplier(attributeValue));
```

Comment:
Is this really necessary? I think we could set attribute here as is, define our own `Supplier` and use it in `SessionResultSetExtractor`. `JdbcSession#getAttribute` would then test if the attribute is instance of the supplier we defined and if yes execute `#get` on it. That would reduce the impact and simplify things.

Reply:
I initially thought of that approach as well, but decided not to go with it. My reasons are:
* I believe `instanceof` checking is generally discouraged in favor of polymorphism (which is what I ended up doing)
* That would prohibit the storage of instances of that special class (as doing so would result in unintended behavior)

Would you still like me to go ahead with that approach?